Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp

```diff
@@ -193,37 +198,38 @@ void SourceBufferPrivateGStreamer::allSamplesInTrackEnqueued(const AtomString& t
 
 void SourceBufferPrivateGStreamer::didReceiveInitializationSegment(InitializationSegment&& initializationSegment)
 {
-    auto initCheckTask = [weakThis = WeakPtr { *this }, this] (InitializationSegment& segment) {
-        if (!weakThis)
-            return false;
-
-        for (auto& trackInfo : segment.videoTracks) {
-            GRefPtr<GstCaps> initialCaps = static_cast<VideoTrackPrivateGStreamer*>(trackInfo.track.get())->initialCaps();
-            ASSERT(initialCaps);
-            if (!m_tracks.contains(trackInfo.track->id()))
-                m_tracks.add(trackInfo.track->id(), MediaSourceTrackGStreamer::create(TrackPrivateBaseGStreamer::TrackType::Video, trackInfo.track->id(), WTFMove(initialCaps)));
-        }
-        for (auto& trackInfo : segment.audioTracks) {
-            GRefPtr<GstCaps> initialCaps = static_cast<AudioTrackPrivateGStreamer*>(trackInfo.track.get())->initialCaps();
-            ASSERT(initialCaps);
-            if (!m_tracks.contains(trackInfo.track->id()))
-                m_tracks.add(trackInfo.track->id(), MediaSourceTrackGStreamer::create(TrackPrivateBaseGStreamer::TrackType::Audio, trackInfo.track->id(), WTFMove(initialCaps)));
-        }
-        for (auto& trackInfo : segment.textTracks) {
-            GRefPtr<GstCaps> initialCaps = static_cast<InbandTextTrackPrivateGStreamer*>(trackInfo.track.get())->initialCaps();
-            ASSERT(initialCaps);
-            if (!m_tracks.contains(trackInfo.track->id()))
-                m_tracks.add(trackInfo.track->id(), MediaSourceTrackGStreamer::create(TrackPrivateBaseGStreamer::TrackType::Text, trackInfo.track->id(), WTFMove(initialCaps)));
-        }
+    SourceBufferPrivate::didReceiveInitializationSegment(WTFMove(initializationSegment));
+}
```

Comment:
Seems like this could just be removed entirely.

Reply:
yes indeed.